js/ui/projectManager.js

```diff
@@ -505,12 +515,66 @@ define([
       if (!keepRetained) chrome.storage.local.remove("retainedProject");
       this.render();
     },
-    
+
     getPaths: function() {
       return Object.keys(this.pathMap);
     }
   };
 
+  // Resize sidebar
+  // min / max sidebar width
+  var MIN_WIDTH = 100;
+  var MAX_WIDTH = 500;
+
+  function setupSidebarResize(element, document) {
+    // Sidebar resizing
+    var isResizing = false;
+    // make sure width is not set in element `style`
+    element.style.width = null;
+
+    var startSidebarResize = function (evt) {
+      if (evt.target.hasClass('project-resizer')) {
+        // do not resize when 'autohide' is on
+        if (element.hasClass('autohide')) return;
+
+        evt.preventDefault();
+        evt.stopPropagation();
+
+        element.addClass('_resize');
+        isResizing = true;
+      }
+    };
+
+    var doSidebarResize = function (evt) {
+      if (!isResizing) return;
+
+      var mouseX = evt.pageX;
+      if (mouseX > MIN_WIDTH && mouseX < MAX_WIDTH) {
+        element.style.width = evt.pageX + 'px';
+      }
+    };
+
+    var stopSidebarResize = function (evt) {
+      if (!isResizing) return;
+
+      evt.preventDefault();
+      evt.stopPropagation();
+
+      element.removeClass('_resize');
+      isResizing = false;
+
+      //Ace doesn't know about non-window resize events
+      //moving the panel will screw up its dimensions
+      setTimeout(function() {
+        editor.resize();
+      }, 100);
+    };
+
+    document.addEventListener('mousedown', startSidebarResize);
+    document.addEventListener('mousemove', doSidebarResize);
+    document.addEventListener('mouseup', stopSidebarResize);
```

Comment:
I think it might be nice to add and remove these listeners on demand, instead of having them available all the time. I know you have the early exit based on `isResizing` in each one, but it's not a bad idea to just cut the function call over head altogether.